src/components/FormSquadScreen.js

```diff
@@ -1,19 +1,57 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 import { View, Pressable, Text, StyleSheet, Platform } from "react-native";
 import DateTimePicker from "@react-native-community/datetimepicker";
 import SelectDropdown from "react-native-select-dropdown";
 import { FlatList, TextInput } from "react-native-gesture-handler";
+import { useNavigation } from "@react-navigation/native";
+import { useFocusEffect } from "@react-navigation/native";
+
+import { postSquad } from "../apiCalls";
 
 const FormSquadScreen = ({ allUsers, userGames }) => {
   const [date, setDate] = useState(new Date());
   const [mode, setMode] = useState("date");
   const [showing, setShowing] = useState(false);
-  const [selected, setSelected] = useState('');
-  const [currentUserGames, setCurrentUserGames] = useState([]);
-  const [users, setUsers] = useState([]);
+  const [selected, setSelected] = useState("");
+  const [users, setUsers] = useState(allUsers);
   const [filterByNameValue, setFilterByNameValue] = useState("");
   const [squadMembers, setSquadMembers] = useState([]);
-  const [squadFull, setSquadFull] = useState(false);
+  const [competitive, setCompetitive] = useState(false);
+  const [squadFull, setSquadFull] = useState(false); // needs full squad error handling
+
+  const navigation = useNavigation();
+
+  const formSquadHandler = () => {
+    const squad = {
+      id: 1,
+      game: selected,
+      eventTime: date.toISOString(),
+      numberPlayers: squadMembers.length,
+      competitive: competitive,
+      squadMembers: squadMembers,
+    };
+    postSquad(squad)
+      .then((response) => {
+        console.log(response);
+        navigation.navigate("My Squads");
+      })
+      .catch((error) => console.log(error));
+  };
+
+  // React Native components don't unmount causing this component not to update then it's navigated to and from,
+  // useFocusEffect tells the component to do stuff when the user navigates to of from the screen
```

Comment:
This should probably specify React Native using navigation

Reply:
So awesome finding useFocusEffect! Thanks, Anna!

Reply:
great find 👍